common/api-review/data-connect.api.md

```diff
@@ -0,0 +1,311 @@
+}
+
+// @public (undocumented)
+export interface DataConnectTransport {
+    // (undocumented)
+    invokeMutation<T, U>(queryName: string, body?: U): PromiseLike<{
+        data: T;
+        errors: Error[];
+    }>;
+    // (undocumented)
+    invokeQuery<T, U>(queryName: string, body?: U): PromiseLike<{
+        data: T;
+        errors: Error[];
+    }>;
+    // (undocumented)
+    onTokenChanged: (token: string | null) => void;
+    // (undocumented)
+    useEmulator(host: string, port?: number, sslEnabled?: boolean): void;
+}
+
+// @public (undocumented)
+export type DataSource = typeof SOURCE_CACHE | typeof SOURCE_SERVER;
+
+// @public
+export function executeMutation<Data, Variables>(mutationRef: MutationRef<Data, Variables>): MutationPromise<Data, Variables>;
+
+// @public
+export function executeQuery<Data, Variables>(queryRef: QueryRef<Data, Variables>): QueryPromise<Data, Variables>;
+
+// @public (undocumented)
+export const FIREBASE_DATA_CONNECT_EMULATOR_HOST_VAR = "FIREBASE_DATA_CONNECT_EMULATOR_HOST";
+
+// @public (undocumented)
+export class FirebaseAuthProvider implements AuthTokenProvider {
+    constructor(_appName: string, _options: FirebaseOptions, _authProvider: Provider<FirebaseAuthInternalName>);
+    // (undocumented)
+    addTokenChangeListener(listener: AuthTokenListener): void;
+    // (undocumented)
+    getToken(forceRefresh: boolean): Promise<FirebaseAuthTokenData | null>;
+    // (undocumented)
+    removeTokenChangeListener(listener: (token: string | null) => void): void;
+}
+
+// @public
+export function getDataConnect(options: ConnectorConfig): DataConnect;
+
+// @public
+export function getDataConnect(app: FirebaseApp, options: ConnectorConfig): DataConnect;
+
+// @public (undocumented)
+export const MUTATION_STR = "mutation";
+
+// @public
+export interface MutationPromise<Data, Variables> extends PromiseLike<MutationResult<Data, Variables>> {
+}
+
+// @public (undocumented)
+export interface MutationRef<Data, Variables> extends OperationRef<Data, Variables> {
+    // (undocumented)
+    refType: typeof MUTATION_STR;
+}
+
+// @public
+export function mutationRef<Data>(dcInstance: DataConnect, mutationName: string): MutationRef<Data, undefined>;
+
+// @public (undocumented)
+export function mutationRef<Data, Variables>(dcInstance: DataConnect, mutationName: string, variables: Variables): MutationRef<Data, Variables>;
+
+// @public (undocumented)
+export interface MutationResponse<T> extends CancellableOperation<T> {
+}
+
+// @public
+export interface MutationResult<Data, Variables> extends DataConnectResult<Data, Variables> {
+    // (undocumented)
+    ref: MutationRef<Data, Variables>;
+}
+
+// @public
+export type OnCompleteSubscription = () => void;
+
+// @public
+export type OnErrorSubscription = (err?: FirebaseError) => void;
+
+// @public
+export type OnResultSubscription<Data, Variables> = (res: QueryResult<Data, Variables>) => void;
+
+// @public (undocumented)
```

Comment:
There's a number of undocumented symbols and methods here, there's no rule against it but double check you dont' want to add any reference documentation for these.

Reply:
Updated. Thanks for catching that